exploration-module/src/main/java/io/edpn/backend/exploration/adapter/persistence/MybatisRingRepository.java

```diff
@@ -0,0 +1,46 @@
+package io.edpn.backend.exploration.adapter.persistence;
+
+import io.edpn.backend.exploration.adapter.persistence.entity.MybatisRingEntity;
+import io.edpn.backend.mybatisutil.UuidTypeHandler;
+import org.apache.ibatis.annotations.Result;
+import org.apache.ibatis.annotations.ResultMap;
+import org.apache.ibatis.annotations.Results;
+import org.apache.ibatis.annotations.Select;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface MybatisRingRepository {
+    
+    @Select({"INSERT INTO ring (id, inner_radius, mass, name, outer_radius, ring_class, body_id, star_id) " +
```

Comment:
I discussed this with Kal and I'm under the impression that a ring can have either a body or a star as its "parent". I don't like the solution, though, and suggested some alternatives. It's a bit of a balancing act and most likely requires a link table. Preferably would allow cascading deletes, but I don't think that should be a hard requirement guiding the design.

Reply:
Can we expand on the benefit of a link table for this?

Wouldn't this just offset the dual key requirement to the link table instead of the ring table?

If instead you make a body_ring table and a star_ring table and link those, then when I want to search rings with the exact same data structure I have to scour two tables instead of one. 

Currently seems to be the most straightforward layout.